RestaurantReservation.Db/Repositories/OrderRepository.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Microsoft.EntityFrameworkCore;
+using RestaurantReservation.Db.Models;
+using System.Linq;
+
+namespace RestaurantReservation.Db.Repositories;
+
+public class OrderRepository : BaseRepository<Order>
+{
+    public OrderRepository(RestaurantReservationDbContext context) 
+        : base(context)
+    {
+    }
+    public List<MenuItem> ListOrderedMenuItems(int reservationId)
+    {
+        return _dbContext.OrderItems
+                        .Where(o => o.Order.ReservationId == reservationId)
+                        .Select(o => o.Item)
+                        .ToList();
+    }
+    public List<OrderItem> ListOrdersAndMenuItems(int reservationId)
+    {
+         return _dbContext.OrderItems
+           .Where(o => o.Order.ReservationId == reservationId)
+           .Include(o => o.Order)
+           .ThenInclude(o => o.OrderItems)
+           .ThenInclude(x => x.Item)
+           .AsSplitQuery()
```

Comment:
I like that you know it, but please use `AsSplitQuery` carefully as it runs multiple queries, which can increase database load. Only use it when necessary to prevent data issues.